src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -9849,6 +9849,215 @@ .maxstack  2
+
+            var v0 = CompileAndVerify(compilation0);
+            v0.VerifyIL("Program.G(int)", @"
+{
+  // Code size       27 (0x1b)
+  .maxstack  1
+  .locals init (int V_0,
+                int V_1,
+                object V_2)
+  IL_0000:  nop
+  IL_0001:  ldarg.0
+  IL_0002:  brfalse.s  IL_0006
+  IL_0004:  br.s       IL_000a
+  IL_0006:  ldc.i4.0
+  IL_0007:  stloc.0
+  IL_0008:  br.s       IL_000e
+  IL_000a:  ldc.i4.1
+  IL_000b:  stloc.0
+  IL_000c:  br.s       IL_000e
+  IL_000e:  ldloc.0
+  IL_000f:  stloc.1
+  IL_0010:  ldloc.1
+  IL_0011:  box        ""int""
+  IL_0016:  stloc.2
+  IL_0017:  br.s       IL_0019
+  IL_0019:  ldloc.2
+  IL_001a:  ret
+}
+");
+            v0.VerifyIL("Program.N(out int)", @"
+{
+  // Code size       10 (0xa)
+  .maxstack  2
+  .locals init (object V_0)
+  IL_0000:  nop
+  IL_0001:  ldarg.0
+  IL_0002:  ldc.i4.1
+  IL_0003:  stind.i4
+  IL_0004:  ldnull
+  IL_0005:  stloc.0
+  IL_0006:  br.s       IL_0008
+  IL_0008:  ldloc.0
+  IL_0009:  ret
+}
+");
+
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, n0, n1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            diff1.VerifySynthesizedMembers();
+
+            diff1.VerifyIL("Program.G(int)", @"
+{
+  // Code size       50 (0x32)
+  .maxstack  2
+  .locals init ([int] V_0,
+                [int] V_1,
+                [object] V_2,
+                int V_3, //x
+                int V_4,
+                int V_5,
+                int V_6,
+                int V_7,
+                object V_8)
+  IL_0000:  nop
+  IL_0001:  ldarg.0
+  IL_0002:  stloc.s    V_4
+  IL_0004:  ldloca.s   V_3
+  IL_0006:  call       ""int Program.N(out int)""
+  IL_000b:  stloc.s    V_6
+  IL_000d:  ldloc.s    V_6
+  IL_000f:  brfalse.s  IL_0013
+  IL_0011:  br.s       IL_0018
+  IL_0013:  ldc.i4.0
+  IL_0014:  stloc.s    V_5
+  IL_0016:  br.s       IL_001d
+  IL_0018:  ldc.i4.1
+  IL_0019:  stloc.s    V_5
+  IL_001b:  br.s       IL_001d
+  IL_001d:  ldloc.s    V_5
+  IL_001f:  stloc.s    V_7
+  IL_0021:  ldloc.s    V_4
+  IL_0023:  ldloc.s    V_7
+  IL_0025:  add
+  IL_0026:  box        ""int""
+  IL_002b:  stloc.s    V_8
+  IL_002d:  br.s       IL_002f
+  IL_002f:  ldloc.s    V_8
+  IL_0031:  ret
+}
+");
+            diff1.VerifyIL("Program.N(out int)", @"
+{
+  // Code size       10 (0xa)
+  .maxstack  2
+  .locals init (int V_0)
+  IL_0000:  nop
+  IL_0001:  ldarg.0
+  IL_0002:  ldc.i4.1
+  IL_0003:  stind.i4
+  IL_0004:  ldc.i4.0
+  IL_0005:  stloc.0
+  IL_0006:  br.s       IL_0008
+  IL_0008:  ldloc.0
+  IL_0009:  ret
+}
+");
+
+            var diff2 = compilation2.EmitDifference(
```

Comment:
@gafter , @333fred , @jcouv 

Here is a crash:


```
Microsoft.CodeAnalysis.CSharp.EditAndContinue.UnitTests.EditAndContinueTests.OutVar_InSwitchExpression FAILED:
	Exception type: 'System.ArgumentException', number: '0', parent: '-1'
	Exception message:
An item with the same key has already been added.
	Exception stacktrace
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.CodeAnalysis.Emit.EncVariableSlotAllocator..ctor(SymbolMatcher symbolMap, Func`2 syntaxMapOpt, IMethodSymbolInternal previousTopLevelMethod, DebugId methodId, ImmutableArray`1 previousLocals, IReadOnlyDictionary`2 lambdaMapOpt, IReadOnlyDictionary`2 closureMapOpt, String stateMachineTypeNameOpt, Int32 hoistedLocalSlotCount, IReadOnlyDictionary`2 hoistedLocalSlotsOpt, Int32 awaiterCount, IReadOnlyDictionary`2 awaiterMapOpt, LambdaSyntaxFacts lambdaSyntaxFacts)
   at Microsoft.CodeAnalysis.Emit.DefinitionMap`1.TryCreateVariableSlotAllocator(EmitBaseline baseline, Compilation compilation, IMethodSymbolInternal method, IMethodSymbol topLevelMethod, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Emit.PEDeltaAssemblyBuilder.TryCreateVariableSlotAllocator(MethodSymbol method, MethodSymbol topLevelMethod, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.LowerBodyOrInitializer(MethodSymbol method, Int32 methodOrdinal, BoundStatement body, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState, Boolean instrumentForDynamicAnalysis, DebugDocumentProvider debugDocumentProvider, ImmutableArray`1& dynamicAnalysisSpans, DiagnosticBag diagnostics, VariableSlotAllocator& lazyVariableSlotAllocator, ArrayBuilder`1 lambdaDebugInfoBuilder, ArrayBuilder`1 closureDebugInfoBuilder, StateMachineTypeSymbol& stateMachineTypeOpt)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.VisitNamedType(NamedTypeSymbol symbol, TypeCompilationState arg)
   at Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.Accept[TArgument,TResult](CSharpSymbolVisitor`2 visitor, TArgument argument)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamespace(NamespaceSymbol symbol)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethodBodies(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean emittingPdb, Boolean emitTestCoverageData, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean emitMetadataOnly, Boolean emitTestCoverageData, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Compilation.Compile(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.Emit.EmitHelpers.EmitDifference(CSharpCompilation compilation, EmitBaseline baseline, IEnumerable`1 edits, Func`2 isAddedSymbol, Stream metadataStream, Stream ilStream, Stream pdbStream, ICollection`1 updatedMethods, CompilationTestData testData, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.EmitDifference(EmitBaseline baseline, IEnumerable`1 edits, Func`2 isAddedSymbol, Stream metadataStream, Stream ilStream, Stream pdbStream, ICollection`1 updatedMethods, CompilationTestData testData, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.EmitDifference(Compilation compilation, EmitBaseline baseline, ImmutableArray`1 edits, IEnumerable`1 allAddedSymbols, CompilationTestData testData)
   at Microsoft.CodeAnalysis.CSharp.EditAndContinue.UnitTests.EditAndContinueTests.OutVar_InSwitchExpression()


```

Reply:
Obviously we cannot integrate a PR that has failing tests.  Please fix the crash.  Alternatively, withdraw this PR or skip this test with a skip reason pointing to an issue describing the problem.

---
In reply to: [300811583](https://github.com/dotnet/roslyn/pull/37025#discussion_r300811583) [](ancestors = 300811583)

Reply:
Thank you, @gafter ! I have added https://github.com/dotnet/roslyn/issues/37047 and skipped the test.